doc/source/whatsnew/v2.1.0.rst

```diff
@@ -775,6 +775,8 @@ Plotting
 
 Groupby/resample/rolling
 ^^^^^^^^^^^^^^^^^^^^^^^^
+- Bug in :meth:`DataFrame.groupby` When using the rank function on grouped data frames, the result is different when using nullable datatypes compared to numpy datatypes.
+  This is only an issue when passing ``na_option="bottom"`` or ``na_option="top"`` (:issue:`54206`)
```

Comment:
I think we can make this a bit more concise. Also we use a script to maintain the sort order in the whatsnew - each note should be on a single line. What do you think of something like:

```
- Bug in :meth:`DataFrame.groupby` when using the rank function on nullable datatypes when passing ``na_option="bottom"`` or ``na_option="top"`` (:issue:`54206`)
```